Algorithm/QCAlgorithm.cs

```diff
@@ -2169,28 +2173,37 @@ public void SetCurrentSlice(Slice slice)
             CurrentSlice = slice;
         }
 
+        /// <summary>
+        /// Sets the order event provider
+        /// </summary>
+        /// <param name="newOrderEvent">The order event provider</param>
+        public void SetOrderEventProvider(IOrderEventProvider newOrderEvent)
+        {
+            newOrderEvent.NewOrderEvent += (sender, orderEvent)
+                => { EmitInsightBasedOnFill(orderEvent); };
+        }
+
         /// <summary>
         /// Will emit a new <see cref="Insight"/> based on the given
         /// <see cref="OrderEvent"/>
         /// </summary>
         /// <remarks>To be called before updating the <see cref="Portfolio"/></remarks>
         /// <param name="orderEvent">The <see cref="OrderEvent"/> from which
         /// the new <see cref="Insight"/> will be created</param>
-        public void EmitInsightBasedOnFill(OrderEvent orderEvent)
+        private void EmitInsightBasedOnFill(OrderEvent orderEvent)
```

Comment:
Not sure if there is a possibility of user algorithms depending on this remaining public? Just don't want to break user builds.

Reply:
This was added recently, just a couple of days ago, and is expected to be called from outside the `QCAlgorithm` (no user calls), the original implementation should probably have been the one in this PR.